### PR TITLE
Fixed chpldoc issue #17110

### PR DESCRIPTION
**Pull request by tt-leader**
Fixed chpldoc incorrect output for an initialization usecase issue #17110  by adding some condition statement of VarSymbol immediate const_kind. 

Also removed the future test file and added some strong testcase in ``variableInitialization.doc.chpl`` Besides, an incorrect output been found in ``calls.doc.chpl`` and corrected.

Tested with chpldoc test and all passed.

### CODE REVIEW

**Comment by tt-leader** [comment]
@lydia-duncan Please help to review the code, and noticed that make_doc CI failed due to chapel compiler internal error.

File: compiler/AST/AstToText.cpp [hunk 1]
```diff
@@ -1469,7 +1469,16 @@ void AstToText::appendExpr(CallExpr* expr, bool printingType, const char *outer,
         if (i > 1)
           mText += ", ";
 
-        appendExpr(expr->get(i), printingType);
+        if (isSymExpr(expr->get(i)) 
+          && isVarSymbol(toSymExpr(expr->get(i))->symbol()) 
+          && toVarSymbol(toSymExpr(expr->get(i))->symbol())->immediate->const_kind == CONST_KIND_STRING)
```
Comment (lydia-duncan) [inline compiler/AST/AstToText.cpp:1474]: I believe you need a check against if the VarSymbol stores an immediate here, using the `isImmediate()` method.  I think that's why you're getting the internal compiler error - VarSymbols that don't store an immediate will have the immediate field be NULL, causing issues when you look at the const_kind of a NULL

Comment (tt-leader) [reply compiler/AST/AstToText.cpp:1474]: Yes, my mistake. I will add ``isImmediate()`` check.

**Comment by lydia-duncan** [comment]
Looks like you missed the DCO on the most recent commit

**Comment by tt-leader** [comment]
Sorry about that, I'm not realized Github web will use another email address, and I fixed it by following command, it's that ok?
```
git commit --amend --no-edit --signoff
git push --force-with-lease origin fixchpldocIssue17110
```

**Comment by lydia-duncan** [comment]
Looks like it worked!

**Comment by lydia-duncan** [comment]
Passed a full round of testing and looks good to me, thanks for the fix!

**Pull request merged**